### PR TITLE
Fix the setting of scalar threshold option

### PR DESCRIPTION
**Pull request by Kai-AMD**

### CODE REVIEW

File: context/llpcPipelineContext.cpp [hunk 1]
```diff
@@ -416,11 +416,24 @@ void PipelineContext::SetOptionsInPipeline(
...
+            shaderOptions.loadScalarizerThreshold = 0;
+            if (EnableScalarLoad)
+            {
+                shaderOptions.loadScalarizerThreshold = ScalarThreshold;
+            }
 #if LLPC_CLIENT_INTERFACE_MAJOR_VERSION >= 33
-            loadScalarizerEnabled |= pShaderInfo->options.enableLoadScalarizer;
+            if (pShaderInfo->options.enableLoadScalarizer)
+            {
+                if (pShaderInfo->options.scalarThreshold!=0)
```
Comment (amdrexu) [inline context/llpcPipelineContext.cpp]: pShaderInfo->options.scalarThreshold!=0
->
pShaderInfo->options.scalarThreshold != 0

File: context/llpcPipelineContext.cpp [hunk 2]
```diff
@@ -416,11 +416,24 @@ void PipelineContext::SetOptionsInPipeline(
...
+                {
+                    shaderOptions.loadScalarizerThreshold = pShaderInfo->options.scalarThreshold;
+                }
+                else
+                {
+                    shaderOptions.loadScalarizerThreshold = 0xFFFFFFFF;
```
Comment (amdrexu) [inline context/llpcPipelineContext.cpp]: static cl::opt<unsigned> ScalarThreshold("scalar-threshold",
                                         cl::desc("The threshold for load scalarizer"),
                                         cl::init(0xFFFFFFFF));

I think you can use InvalidValue. If it couldn't express the meaning of "max". You can define: static const uint32_t MaxScalarThreshold = 0xFFFFFFFF.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 497d0182294d30fd1ad5389b617d2f6cbc8e6dd6
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 40af910391fb8c287cb37bf520c41310bf88d405</li>
<li>SPVGEN: dev ce06cb5e3116ba77a22c3278dfeadfd865a8977c</li>
<li>XGL: dev 7e13a8bd0bb57d3cfb3bc014f6b26a8c9bb8bfd9</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev c22b8992220d645b9f781e4a47947ae546fcd11c</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 939/939 (100.0%)</li>
<li>Failed: 0/939 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 65433/65433 (100.0%)</li>
<li>Failed: 0/65433 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 65465/65465 (100.0%)</li>
<li>Failed: 0/65465 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 65465/65465 (100.0%)</li>
<li>Failed: 0/65465 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 1d54521e402d3495e92e45e7fd7c4ba9864eab62
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 40af910391fb8c287cb37bf520c41310bf88d405</li>
<li>SPVGEN: dev ce06cb5e3116ba77a22c3278dfeadfd865a8977c</li>
<li>XGL: dev 7e13a8bd0bb57d3cfb3bc014f6b26a8c9bb8bfd9</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev c22b8992220d645b9f781e4a47947ae546fcd11c</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 939/939 (100.0%)</li>
<li>Failed: 0/939 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 65433/65433 (100.0%)</li>
<li>Failed: 0/65433 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 65465/65465 (100.0%)</li>
<li>Failed: 0/65465 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 65465/65465 (100.0%)</li>
<li>Failed: 0/65465 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 47d5df123eadb7f153c9d817e5f99976b3a9ec3e
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 40af910391fb8c287cb37bf520c41310bf88d405</li>
<li>SPVGEN: dev ce06cb5e3116ba77a22c3278dfeadfd865a8977c</li>
<li>XGL: dev 7e13a8bd0bb57d3cfb3bc014f6b26a8c9bb8bfd9</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev c22b8992220d645b9f781e4a47947ae546fcd11c</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 939/939 (100.0%)</li>
<li>Failed: 0/939 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 65433/65433 (100.0%)</li>
<li>Failed: 0/65433 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 65465/65465 (100.0%)</li>
<li>Failed: 0/65465 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 65465/65465 (100.0%)</li>
<li>Failed: 0/65465 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**